timesketch/lib/analyzers/contrib/hashlookup_analyzer.py

```diff
@@ -0,0 +1,152 @@
+        self.total_event_counter = 0
+        self.request_set = set()
+
+    @staticmethod
+    def get_kwargs():
+        """Get kwargs for the analyzer.
+
+        Returns:
+            Info to connect to Hashlookup.
+        """
+
+        hashlookup_url = current_app.config.get("HASHLOOKUP_URL")
+
+        if not hashlookup_url:
+            logger.error("Hashlookup conf not found")
+            return []
+
+        matcher_kwargs = [{"hashlookup_url": hashlookup_url}]
+        return matcher_kwargs
+
+    def get_hash_info(self, hash_value):
+        """Search event on Hashlookup.
+
+        Returns:
+            JSON of Hashlookup's results
+        """
+        results = requests.get(f"{self.hashlookup_url}sha256/{hash_value}")
+
+        if not "message" in results.json() and results.status_code != 200:
+            logger.error("Error with Hashlookup url")
+            return []
+        if "message" in results.json():
+            return []
+
+        self.total_event_counter += 1
+
+        return results.json()
+
+    def mark_event(self, event, hash_value):
+        """Annotate an event with data from Hashlookup
+
+        Tags with validate emoji, adds a comment to the event.
+        """
+
+        event.add_comment(f"{self.hashlookup_url}sha256/{hash_value}")
+
+        event.add_tags(["Hashlookup"])
+        event.add_emojis([emojis.get_emoji("VALIDATE")])
+        event.commit()
+
+    def query_hashlookup(self, query, return_fields):
+        """Get event from timesketch, request Hashlookup and mark event."""
+
+        events = self.event_stream(query_string=query, return_fields=return_fields)
+        create_a_view = False
+        for event in events:
+            hash_value = None
+            for key in return_fields:
+                if key in event.source.keys():
+                    hash_value = event.source.get(key)
+                    break
+            if not hash_value:
+                error_hash_counter += 1
+                logger.warning(
+                    "No matching field with a hash found in this event! "
+                    "-- Event Source: %s",
+                    str(event.source),
+                )
+                continue
+
+            if len(hash_value) != 64:
+                logger.warning(
+                    "The extracted hash does not match the required "
+                    "lenght (64) of a SHA256 hash. Skipping this "
+                    "event! Hash: %s - Lenght: %d",
+                    hash_value,
+                    len(hash_value),
+                )
+                error_hash_counter += 1
+                continue
+
+            if not hash_value in self.request_set:
+                result = self.get_hash_info(hash_value)
+                if result:
+                    create_a_view = True
+                    self.mark_event(event, hash_value)
+                self.request_set.add(hash_value)
+            else:
```

Comment:
This will mark events containing the hash as soon as they're stored, no? e.g. if a hash is queried, and no results are returned, it's skipped and not marked, but the second time we run into this hash it will be in  `self.request_set` and thus marked.

Reply:
Fixed in 18325158858d824839fb5825bcdc6e1bd6c14552.